### ISSUE TITLE
The TOC of the Clarity wizard must indicate state of success with means different than color

### ISSUE THREAD

**Issue by bashibozuk** [vmware/clarity#3976]
## Summary

The completion of steps in Clarity wizard currently is indicated only with changed color in the TOC which is not readable by screen readers or accessible for people with low visual

* What is the change?
For each completed step in the step navigation, apply off-screen text or use aria-label to describe the completed state of the step.
* Why should it go in Clarity?
To ensure basic accessibility compliance
* Does this change impact existing behaviors? If so how?
No
* If this change introduces a new behavior, is this behavior accessible?
Yes

## Use case

As a screen reader user of Clarity components I want to be notified about the completion state of the steps in the Clarity wizard

## Examples

N/A
## Workarounds

None

**Comment by gnomeontherun** [comment]
@chlane Did you review this part of the wizard or have feedback here? There are demos on the website to review, but I believe we're talking about the green highlight color applied to the left of the step button when a step is completed.

**Comment by chlane** [comment]
Yes I looked at this and thought that aria-selected on the current step was good enough to let screen reader users know where they are in the wizard and that the green bar was more decorative than informative in the context of a wizard because the current step is visually styled with CSS and programatically conveyed with aria-selected.

But, this issue made be double check and I found that our implementation of aria-selected is not working with Firefox/NVDA. The cause is most likely that we are putting the aria-selected property on the wrong element. ARIA properties should applied to elements we want to apply them to, which are typically elements with interactive roles.

Currently, we are placing aria-selected on the <div> that wraps the button as seen below.

`<div class="clr-wizard-stepnav-item clr-nav-link nav-item active ng-star-inserted" clr-wizard-stepnav-item="" ng-reflect-page="[object Object]" id="clr-wizard-step-0" aria-selected="true" aria-controls="clr-wizard-step-0">`

<button class="btn btn-link clr-wizard-stepnav-link" type="button">

But, we should be using the ARIA propertes on the <button> element as seen below.

```
<div class="clr-wizard-stepnav-item clr-nav-link nav-item active ng-star-inserted" clr-wizard-stepnav-item="" ng-reflect-page="[object Object]" id="clr-wizard-step-0">

<button aria-selected="true" aria-controls="clr-wizard-step-0" class="btn btn-link clr-wizard-stepnav-link" type="button">

```

Also, in retrospect I'm recommending aria-current over aria-selected.

See https://www.w3.org/TR/wai-aria-1.1/#aria-current

`<button aria-current="step" aria-controls="clr-wizard-step-0" class="btn btn-link clr-wizard-stepnav-link" type="button">`

That being said, I think the green link is redundant/decorative in the wizard context where completion of previous steps is assumed I'm not against adding offscreen text or an aria-label to the completed steps to say they are completed in case the screen reader user decides to go back and read these controls but I don't think it is necessary.

**Comment by bashibozuk** [comment]
@chlane I am not sure how aria-selected , which indicates the current selection , i.e. the page that we are now on is related to how previous, successfully passed page is indicated. Could you explain please ?

**Comment by chlane** [comment]
@bashibozuk I was thinking that the user would understand that that each step was dependent on the completion of the previous the step and that identifying the current step is good enough. But as I said in my previous comment I'm not against adding off-screen text or an aria-label to the completed steps to say they are completed in case the screen reader user decides to go back and read these controls.  I recommend that we 1) fix the placement of aria-selected 2) use off-screen-text or aria-label to convey the completed state of previous steps.

**Comment by chlane** [comment]
The stepper component also used the green bar to convey the previous step. I recommend to use use off-screen-text or aria-label to convey the completed state of previous steps in the case of the stepper component as well. But do not use aria-current="step" on the stepper component since aria-expanded better represents the behavior of disclosures (https://www.w3.org/TR/wai-aria-practices-1.1/#disclosure). In these cases, the expanded content is implicitly current.

**Comment by brian-free** [comment]
This seems to apply here. We tried to add check icons for the completed steps with CSS but not very successfully to give more than the "green" as a visual indicator.
![image](https://user-images.githubusercontent.com/43282191/70665691-534d4b00-1c2a-11ea-82f9-3dbd60479c1c.png)

**Comment by chlane** [comment]
Yes it looks like there are redundant visual indicators for step completion, the light grey text and the checkbox. The visual redundancy is not  an accessibility issue but we need to provide an aria-label or offscreen text to tell screen readers these steps are completed.

**Comment by bashibozuk** [comment]
Is there any planned effort for adding non visual indicators for the page states or you recommend using the suggested workaround?

**Comment by clane** [comment]
As I recommended earlier, please use off-screen-text or aria-label to convey the completed state of previous steps.

**Comment by lf-collins** [comment]
I saw the following suggestions related to this problem from LevelAccess, they suggest adding a non-color dependent visual indicator.
> Provide a textual description of each state (completed, current, etc) within the step navigation. This may be done by adding off-screen text describing the status of each step to each navigation element in the step navigation.
> In addition, a non-color dependent visual indicator, must be used to convey information that currently depends only on color. For example, underlining all steps the user has previously visited will provide a non-color dependent visual distinction between steps that the user has visited and steps that have not been started.

**Comment by gnomeontherun** [comment]
Clarity cares deeply about accessibility, and we work with our accessibility team to identify and triage issues. We have an internal tracking system to identify and verify accessibility issues, which has captured this already. Some issues with Clarity Angular are difficult to solve without causing side effects or breaking changes, and we have to balance concerns with fixing issues and breaking changes. As we build out Clarity Core, we are able to address these root cause accessibility issues and recommend adopting the Clarity Core implementations as they become available. We are closing this issue as we have the issues tracked internally. Please follow our development and releases to see when we release relevant Clarity Core components to make this possible.
